BaseTools/Conf/build_rule.template

```diff
@@ -331,7 +331,8 @@
         $(DEBUG_DIR)(+)$(MODULE_NAME)
 
     <Command.MSFT, Command.INTEL, Command.CLANGPDB>
-        "$(DLINK)" $(DLINK_FLAGS) $(DLINK_SPATH) @$(STATIC_LIBRARY_FILES_LIST)
+        "$(CC)" /Fo$$(OUTPUT_DIR)/Empty_C_File_Host_Application_Build.obj $(CC_FLAGS) $(INC) $(EDK_TOOLS_PATH)/Conf/Empty_C_File_Host_Application_Build.c
+        "$(DLINK)" $(DLINK_FLAGS) $(DLINK_SPATH) $(OUTPUT_DIR)/Empty_C_File_Host_Application_Build.obj @$(STATIC_LIBRARY_FILES_LIST)

```

Comment:
Do we need to split off `USER_DEFINED` and `HOST_APPLICATION` for this approach? Will we get some weird compilation for `USER_DEFINED` files with this?

Reply:
I thought about that.  Adding an empty C file to the build if USER_DEFINED should not cause any side effects.  I have not seen any so far.

Reply:
The trick just being that `HOST_APPLICATION` has different `CC_FLAGS` set?